src/llm/io_processing/output_parser.cpp

```diff
@@ -104,7 +105,20 @@ ParsedOutput OutputParser::parse(const std::vector<int64_t>& generatedTokens, co
     return parsedOutput;
 }
 
-std::optional<rapidjson::Document> OutputParser::parseChunk(const std::string& chunkResponse, const bool toolsAvailable) {
+static inline bool isParsingTagPartOfChunk(const std::string& chunk, const std::string& parsingTag) {
+    return chunk.find(parsingTag) != std::string::npos;
+}
+
+static inline bool isAnyOfBeginningOnlyParsingTagPartOfChunk(const std::string& chunk, const std::unordered_set<std::string>& beginningOnlyParsingTags) {
```

Comment:
`beginningOnly` name does not speak to me to be honest

Reply:
to be neither, any ideas?

Reply:
`bool chunkContainsSpecialStartTag(const std::string& chunk, const std::unordered_set<std::string>& specialStartTags)` ?
I keep thinking about the naming for those additional start tags. "special", "secondary" or something else. I think we should give them a name so we can differentiate them from the regular start tag and then base method and variables naming on that.